fe/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -542,13 +542,49 @@ private static DateTimeFormatterBuilder formatBuilder(String pattern) throws Ana
         return builder;
     }
 
-    public DateLiteral plusDays(int day) throws AnalysisException {
+    public enum PlusType {
+        YEAR,
+        MONTH,
+        DAY,
+        HOUR,
+        MINUTE,
+        SECOND
+    }
+
+    public DateLiteral plus(int interval, PlusType plusType) throws AnalysisException{
```

Comment:
Why not implement this function as `plusYear`, `plusMonth`...?
If we do that way, the enum defination and swich clause can be eliminated.